### PR TITLE
Convert okcurl to Kotlin

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
```
yuri@Darias-MBP ~/w/okhttp> java -jar ./okcurl/build/libs/okcurl-4.0.0-SNAPSHOT.jar -A "XXXX" -k --frames https://facebook.com/robots.txt
>> CONNECTION 505249202a20485454502f322e300d0a0d0a534d0d0a0d0a
>> 0x00000000     6 SETTINGS
>> 0x00000000     4 WINDOW_UPDATE
<< 0x00000000    30 SETTINGS
<< 0x00000000     4 WINDOW_UPDATE
>> 0x00000003    34 HEADERS       END_STREAM|END_HEADERS
>> 0x00000000     0 SETTINGS      ACK
<< 0x00000000     0 SETTINGS      ACK
<< 0x00000003     4 WINDOW_UPDATE
<< 0x00000003   422 HEADERS       END_HEADERS
<< 0x00000003   605 DATA
<< 0x00000003    10 DATA          END_STREAM
# Notice: Crawling Facebook is prohibited unless you have express written
# permission. See: http://www.facebook.com/apps/site_scraping_tos_terms.php

User-agent: Applebot
```

File: build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ import net.ltgt.gradle.errorprone.CheckSeverity
 
 buildscript {
   ext.versions = [
-      'airline': '0.8',
+      'airline': '2.6.0',
```
Comment (yschimke) [inline build.gradle]: I had issues with Airline previously with mutable lists in Kotlin, so followed a known working approach.

File: okcurl/src/main/java/okhttp3/curl/Main.kt [hunk 2]
```diff
@@ -13,291 +13,275 @@
...
+
+    private fun versionString(): String {
+      try {
+        val prop = Properties()
+        Main::class.java.getResourceAsStream("/okcurl-version.properties").use {
+          prop.load(it)
+        }
+        return prop.getProperty("version", "dev")
+      } catch (e: IOException) {
+        throw AssertionError("Could not load okcurl-version.properties.")
```
Comment (swankjesse) [inline okcurl/src/main/java/okhttp3/curl/Main.kt]: omit this now-unnecessary try/catch clause?

File: okcurl/src/main/java/okhttp3/curl/Main.kt [hunk 3]
```diff
@@ -13,291 +13,275 @@
...
 
-      @Override public X509Certificate[] getAcceptedIssuers() {
-        return new X509Certificate[0];
+    private fun createInsecureSslSocketFactory(trustManager: TrustManager): SSLSocketFactory {
+      try {
+        val context = Platform.get().newSSLContext()
+        context.init(null, arrayOf(trustManager), null)
+        return context.socketFactory
+      } catch (e: Exception) {
+        throw AssertionError(e)
```
Comment (swankjesse) [inline okcurl/src/main/java/okhttp3/curl/Main.kt]: omit this catch clause?

**Pull request merged**